src/main/java/com/_up/megastore/data/repositories/IProductRepository.java

```diff
@@ -1,17 +1,17 @@
 package com._up.megastore.data.repositories;
 
 import com._up.megastore.data.model.Product;
+import java.util.List;
+import java.util.UUID;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 
-import java.util.List;
-import java.util.UUID;
-
 public interface IProductRepository extends JpaRepository<Product, UUID> {
-    boolean existsByProductIdAndDeletedTrue(UUID productId);
     Page<Product> findProductsByDeletedIsFalseAndVariantOfIsNullAndNameContainingIgnoreCase(String name, Pageable pageable);
-    boolean existsByNameIgnoreCaseAndDeletedIsFalse(String name);
+    Page<Product> findProductsByDeletedIsFalseAndNameContainingIgnoreCase(String name,
+        Pageable pageable);
```

Comment:
Mismo comentario que recien

```suggestion
    Page<Product> findProductsByDeletedAndNameContainingIgnoreCase(boolean deleted, String name,
        Pageable pageable);
```